packages/app/src/cli/services/context.ts

```diff
@@ -331,6 +344,30 @@ export async function ensureDeployContext(options: DeployContextOptions): Promis
   return result
 }
 
+export async function ensureReleaseContext(options: ReleaseContextOptions): Promise<ReleaseContextOutput> {
+  const token = await ensureAuthenticatedPartners()
+  const [partnersApp, envIdentifiers] = await fetchAppAndIdentifiers(options, token)
+  const identifiers: Identifiers = envIdentifiers as Identifiers
+
+  if (!partnersApp.betas?.unifiedAppDeployment) {
```

Comment:
Can we instead use `deploymentMode` instead? Its values can be `'legacy' | 'unified' | 'unified-skip-release'`

Reply:
`deploymentMode` is a local variable [included](https://github.com/Shopify/cli/blob/63dc8c18426d35e2a817ebeaad5043948bb153fd/packages/app/src/cli/services/context.ts#L297) in the method `ensureDeployContext`.  
In the release context, the requirement about the `deploymentMode` is that the command should exit in case the app is configured as `legacy`. Thus there are some differences between the deployment context and the release one:

- `unified-skip-release` - This does not make sense in the `release` context, as we are indeed releasing :) 
- In the deploy context, in case the app is configured as ` legacy` there is a prompt to switch to `unified`. I guess that in the release flow this does not make sense, because a previous version tag in needed to create a release so at this point the app should be using `unified` deployments

Maybe it would be clearer if we do something like this:

```ts
  const deploymentMode: DeploymentMode = partnersApp.betas?.unifiedAppDeployment ? 'unified': 'legacy'
  if (deploymentMode === 'unified') {
    throw new AbortSilentError()
  }
```

however that local variable is going to be used only once in the next conditional